light-prover-sdk/tests/genericProver.test.ts

```diff
@@ -0,0 +1,353 @@
+
+import { it } from "mocha";
+const chai = require("chai");
+const chaiAsPromised = require("chai-as-promised");
+
+// Load chai-as-promised support
+chai.use(chaiAsPromised);
+import {
+  FEE_ASSET,
+  Provider as LightProvider,
+  MINT,
+  Transaction,
+  TransactionParameters,
+  Action,
+  Relayer,
+  Utxo,
+  Account,
+  MerkleTree,
+  IDL_VERIFIER_PROGRAM_ZERO,
+} from "../../light-zk.js/src";
+import { ProofInputs } from "../src/generics";
+
+process.env.ANCHOR_PROVIDER_URL = "http://127.0.0.1:8899";
+process.env.ANCHOR_WALLET = process.env.HOME + "/.config/solana/id.json";
+
+describe("Test Prover Functional", () => {
+  let seed32 = new Uint8Array(32).fill(1).toString();
+  let depositAmount = 20_000;
+  let depositFeeAmount = 10_000;
+
+  let mockPubkey = SolanaKeypair.generate().publicKey;
+  let mockPubkey2 = SolanaKeypair.generate().publicKey;
+  let mockPubkey3 = SolanaKeypair.generate().publicKey;
+  let poseidon,
+    lightProvider: LightProvider,
+    deposit_utxo1,
+    relayer,
+    keypair,
+    paramsDeposit,
+    paramsWithdrawal;
+
+  before(async () => {
+    poseidon = await circomlibjs.buildPoseidonOpt();
+    // TODO: make fee mandatory
+    relayer = new Relayer(
+      mockPubkey3,
+      mockPubkey,
+      mockPubkey,
+      new anchor.BN(5000),
+    );
+    keypair = new Account({ poseidon: poseidon, seed: seed32 });
+    lightProvider = await LightProvider.loadMock();
+    deposit_utxo1 = new Utxo({
+      poseidon: poseidon,
+      assets: [FEE_ASSET, MINT],
+      amounts: [new anchor.BN(depositFeeAmount), new anchor.BN(depositAmount)],
+      account: keypair,
+      blinding: new anchor.BN(new Array(31).fill(1)),
+      assetLookupTable: lightProvider.lookUpTables.assetLookupTable,
+      verifierProgramLookupTable:
+        lightProvider.lookUpTables.verifierProgramLookupTable,
+    });
+    paramsDeposit = new TransactionParameters({
+      outputUtxos: [deposit_utxo1],
+      transactionMerkleTreePubkey: mockPubkey2,
+      lookUpTable: lightProvider.lookUpTable,
+      poseidon,
+      senderSpl: mockPubkey,
+      senderSol: lightProvider.wallet?.publicKey,
+      action: Action.SHIELD,
+      transactionNonce: 0,
+      verifierIdl: IDL_VERIFIER_PROGRAM_ZERO,
+    });
+    lightProvider.solMerkleTree!.merkleTree = new MerkleTree(18, poseidon, [
+      deposit_utxo1.getCommitment(poseidon),
+    ]);
+
+    assert.equal(
+      lightProvider.solMerkleTree?.merkleTree.indexOf(
+        deposit_utxo1.getCommitment(poseidon),
+      ),
+      0,
+    );
+    paramsWithdrawal = new TransactionParameters({
+      inputUtxos: [deposit_utxo1],
+      transactionMerkleTreePubkey: mockPubkey2,
+      poseidon,
+      recipientSpl: mockPubkey,
+      recipientSol: lightProvider.wallet?.publicKey,
+      action: Action.UNSHIELD,
+      relayer,
+      transactionNonce: 0,
+      verifierIdl: IDL_VERIFIER_PROGRAM_ZERO,
+    });
+  });
+
+  it("prover functional test1", async () => {
+    let tx = new Transaction({
```

Comment:
ah now I get why you added the before statement.
I think we should not test transaction compatibility here but in zk.js, because zk.js imports this sdk. Actually I am not sure why this even works since you will have a circular dependency between zk.js and the prover-sdk.

- please test the generic prover for a non light protocol circuit
- remove the light protocol specific tests this should be covered in zk.js tests